src/libraries/System.Private.CoreLib/src/System/IO/PathInternal.cs

```diff
@@ -102,40 +102,90 @@ internal static bool AreRootsEqual(string? first, string? second, StringComparis
         }
 
         /// <summary>
-        /// Try to remove relative segments from the given path (without combining with a root).
+        /// Tries to remove relative segments from the given path, starting after the path root.
```

Comment:
Why is it important to say "starting after the path root" in the description? How would the API behavior differ if it did not start after the path root?

Reply:
It would potentially eat into the "root" section of the path. You don't want to eat the `C:\` for `C:\..` for example.